citadel/indico_citadel/backend.py

```diff
@@ -0,0 +1,180 @@
+            }),
+            ContributionRecordSchema(context={
+                'categories': self.categories,
+                'schema': url_join(self.search_app, 'schemas/indico/contributions_v1.0.0.json')
+            }),
+            SubContributionRecordSchema(context={
+                'categories': self.categories,
+                'schema': url_join(self.search_app, 'schemas/indico/subcontributions_v1.0.0.json')
+            }),
+            AttachmentRecordSchema(context={
+                'categories': self.categories,
+                'schema': url_join(self.search_app, 'schemas/indico/attachments_v1.0.0.json')
+            }),
+            EventNoteRecordSchema(context={
+                'categories': self.categories,
+                'schema': url_join(self.search_app, 'schemas/indico/notes_v1.0.0.json')
+            })
+        ]
+
+    def dump_record(self, obj):
+        for schema in self.schemas:
+            if isinstance(obj, schema.Meta.model):
+                return schema.dump(obj)
+        raise ValueError(f'unknown object ref: {obj}')
+
+    def upload_record(self, entry, entries, session, from_queue):
+        change_type = entries[entry] if from_queue else SimpleChange.created
+        json = self.dump_record(entry)
+        entry_type = get_entry_type(entry)
+        response = None
+        if change_type & SimpleChange.created:
+            response = session.post(self.endpoint_url, json=json)
+        else:
+            search_id = CitadelSearchAppIdMap.get_search_id(entry.id, entry_type)
+            if search_id is None:
+                raise Exception('SearchMapId does not exist for the object')
+
+            if change_type & SimpleChange.updated:
+                url = url_join(self.search_app, f'api/record/{search_id}')
+                response = session.put(url, json=json)
+            elif change_type & SimpleChange.deleted:
+                url = url_join(self.search_app, f'api/record/{search_id}')
+                response = session.delete(url, json=json)
+
+        if not response.ok:
+            raise Exception(f'{response.status_code} - {response.text} in record {json}')
+        elif change_type & SimpleChange.created:
+            content = response.json()
+            CitadelSearchAppIdMap.create(content['metadata']['control_number'], entry.id, entry_type)
+        response.close()
+
+    def upload_file(self, entry, entries, session):
+        with entry.attachment.file.open() as file:
+            response = session.put(
+                url_join(self.search_app, f'api/record/{entry.search_id}/files/attachment'),
+                data=file
+            )
+        if response.ok:
+            entry.attachment_file_id = entry.attachment.file.id
+            db.session.merge(entry)
+            db.session.commit()
+        else:
+            self.logger.error('Failed uploading attachment %d: %s', entry.attachment.id, response.text)
```

Comment:
I just noticed that we log an error here, but in case of `upload_record` we raise an exception. Is this different behavior intentional? Would it make sense to "just" log in both cases? Or always fail since with the retry logic we already only get the error if the error is persistent enough to still happen after all the retries?